kernel/filesystem/FileCache.cpp

```diff
@@ -3,38 +3,45 @@
 #include <arch/Platform.h>
 #include <debug/Log.h>
 #include <memory/Pmm.h>
+#include <memory/virtual/VfsVmDriver.h>
 #include <UnitConverter.h>
+#include <Memory.h>
 #include <Maths.h>
 
 namespace Npk::Filesystem
 {
-    size_t cacheUnitSize;
-    
+    FileCacheInfo cacheInfo;
+
     void InitFileCache()
     {
         //find the smallest granularity supported by the MMU, we'll use that as a cache unit size.
-        cacheUnitSize = -1ul;
+        cacheInfo.unitSize = -1ul;
         const HatLimits& mmuLimits = GetHatLimits();
         for (size_t i = 0; i < mmuLimits.modeCount; i++)
         {
-            if (mmuLimits.modes[i].granularity < cacheUnitSize)
-                cacheUnitSize = mmuLimits.modes[i].granularity;
+            if (mmuLimits.modes[i].granularity < cacheInfo.unitSize)
+            {
+                cacheInfo.unitSize = mmuLimits.modes[i].granularity;
+                cacheInfo.hatMode = i;
+                cacheInfo.modeMultiple = 4;
+            }
         }
-        ASSERT(sl::IsPowerOfTwo(cacheUnitSize), "Bad cache unit size");
+        ASSERT(cacheInfo.unitSize != -1ul, "Unable to select cache unit size");
```

Comment:
so if unitSize changes from -1ul it reports an error?

Reply:
Haha the logic is backwards, assert checks that the condition is true, otherwise it panics with that message.
So here were checking unitSize was actually set to something (and isn't it's initial value).